### PR TITLE
Change the test to use the event watch for accurate assertion 

### PR DESCRIPTION
**Pull request by vatsalparekh**
With the previous way to hit API and do assertion against it might make the change of labels lost in between of the assertions; Switching to use event watch on node this way will make it more reliable
One of the failure: https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/5514/pull-kubevirt-e2e-k8s-1.19/1385290108260847616

Signed-off-by: Vatsal Parekh <vparekh@redhat.com>

**What this PR does / why we need it**:
Fixes the flake for `test_id:4716`

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by fgimenez — COMMENTED** [review]
Could you please add a link to one of the failed builds in the PR description? This one for instance https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/5514/pull-kubevirt-e2e-k8s-1.19/1385290108260847616

It would be good to explain what does this timeout increase mean, for what are we giving more time? Also have some reasoning about why did you decide to increase to 120 seconds instead of any other value.

**Comment by fgimenez** [comment]
/lgtm

Thanks!

**Comment by vatsalparekh** [comment]
This might be the best solution here to fix the flake permanently 
/hold

File: tests/vmi_lifecycle_test.go [hunk 1]
```diff
@@ -110,16 +111,39 @@ var _ = Describe("[rfe_id:273][crit:high][vendor:cnv-qe@redhat.com][level:compon
...
-			err = virtClient.CoreV1().Pods(handlerNamespace).Delete(context.Background(), pod.Name, metav1.DeleteOptions{})
-			Expect(err).ToNot(HaveOccurred())
 
-			Eventually(func() string {
-				n, err := virtClient.CoreV1().Nodes().Get(context.Background(), pod.Spec.NodeName, metav1.GetOptions{})
-				Expect(err).ToNot(HaveOccurred())
+			By("setting up a watch on Nodes")
+			nodeWatch, err := virtClient.CoreV1().Nodes().Watch(context.Background(), metav1.ListOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			signalSchedulableNode := func(stopCn <-chan bool, eventsCn <-chan k8sWatch.Event, scheduleSignalCn chan<- bool) {
```
Comment (qinqon) [inline tests/vmi_lifecycle_test.go:118]: Does it make sense to use gomega channel matchers for this instead of the for/case

Comment (vatsalparekh) [reply tests/vmi_lifecycle_test.go:118]: @qinqon I don't understand
for/case is used to catch events from the channels when it is run as goroutine
We already do the channel assertion at the end using gomega matcher

**Review by qinqon — APPROVED** [review]

**Review by enp0s3 — APPROVED** [review]
/approve

**Pull request merged**